gtsfm/ui/dot_graph_generator.py

```diff
@@ -0,0 +1,108 @@
+"""
+Creates a DOT graph based on the contents of RegistryHolder.REGISTRY.
+
+Author: Kevin Fu
+"""
+
+from gtsfm.ui.registry import RegistryHolder
+
+from pathlib import Path
+import os
+
+import pydot
+from collections import namedtuple
+from inspect import isabstract
+
+JS_ROOT = os.path.join(Path(__file__).resolve().parent.parent.parent, "rtf_vis_tool")
+
+
+class DotGraphGenerator:
+    """Generates and saves a graph of all the components in REGISTRY."""
+
+    def __init__(self, test_mode=False):
+        """Create DotGraphGenerator.
+
+        Args:
+            test_mode - boolean flag, only set True when unit testing.
+        """
+
+        # create empty directed graph
+        self._graph = pydot.Dot("my_graph", graph_type="digraph", bgcolor="white")
+        # TODO: fontname seems broken?
+
+        # style constants, see http://www.graphviz.org/documentation/
+        style_consts = {
+            "blue_fillcolor": "lightskyblue",
+            "gray_fillcolor": "gainsboro",
+            "node_style": '"rounded, filled, solid"',
+            "node_shape": "box",
+            "arrow_color": "gray75",
+        }
+        # namedtuple here is faster than default Dict and allows use of dot
+        # operator
+        StyleTuple = namedtuple("StyleTuple", style_consts)
+        self._style = StyleTuple(**style_consts)
+
+        self._test_mode = test_mode
+
+    def _build_graph(self):
+        """Build graph based on RegistryHolder's REGISTRY."""
+
+        # TODO: remove this
+        print("!\n" * 100)
+        print(RegistryHolder.get_registry())
+
+        for cls_name, cls_type in RegistryHolder.get_registry().items():
+            # don't add the base class to the graph
```

Comment:
consider making a function. You should aim (yet not achieve) for functions/methods <= 4 lines long